django/core/mail/message.py

```diff
@@ -209,6 +209,7 @@ def __init__(
         from_email=None,
         to=None,
         bcc=None,
+        provider=None,
```

Comment:
1. Shouldn't change existing arg order; suggest kwarg only.
2. But: I think the `provider` arg actually belongs in EmailMessage.send(), not in the constructor.

I think it was a mistake to include `connection` as an EmailMessage constructor arg. It really only applies when it's time to send the message. To demonstrate the problem, which server will be used to send these messages?

```python
from django.core.mail import get_connection, EmailMessage

conn1 = get_connection(host="smtp.gmail.com", ...)
msg1 = EmailMessage(connection=conn1, ...)
msg2 = EmailMessage(connection=conn1, ...)

conn2 = get_connection(host="smtp.example.org", ...)
conn2.send_messages([msg1, msg2])
```

We can't easily fix this for with EmailMessage.connection right now, but we should avoid extending it to the new `provider` option.

Reply:
ad 1) fixed
ad 2) as long as `connection` is part of the constructor, `provider` is the expected successor.

Reply:
Well, I don't think `connection` belongs in the constructor either. (It should have been an arg to EmailMessage.send().) `provider` could still be the successor, just moved to the method where it applies.

If we really want to add `provider` to the constructor, I think we should document that it will be ignored when [using `connection.send_messages()` as described in the docs](https://docs.djangoproject.com/en/5.0/topics/email/#sending-multiple-emails).